pkg/build/pipelines/ruby/require.yaml

```diff
@@ -0,0 +1,85 @@
+name: Test a Ruby package require, with optional load clause
+
+needs:
+  packages:
```

Comment:
We should add Ruby here, passed as an input. See go/build for what I'm talking about

Reply:
I just checked the other Ruby pipelines and it seems we didn't include the `Ruby`, was wondering if there was any reason. 🤔

Reply:
I don't think it'd ever been considered that pipeline should fetch the toolchain. `go/build` didn't offer this functionality until a few months ago either. We should provide because it's helpful, and we can implicitly pull in ruby without defining it in the environment

Reply:
That makes sense, I've added Ruby here and couldn't pass the CI error: https://github.com/chainguard-dev/melange/actions/runs/10788637898/job/29919880644?pr=1483#step:9:1071 - for some reason, it doesn't find `ruby-3.2` even its there.

Reply:
So , yeah... several of the pipelines that I've added have this flaw in that they don't declare `needs` that it'd be nice if they did. [`py/pip-build-install`](https://github.com/wolfi-dev/os/blob/main/pipelines/py/pip-build-install.yaml) for example needs bot a pip and a python, but doesn't declare either.

The reason is that it takes `python` as an input (the python version to use).

I don't think  you can use a input to a pipeline in its `needs`. It would be cleaner here if you could.  This specific case would be nice in that you could then:
```
needs:
  packages:
    - ruby-${{inputs.version}}
```

And then have a inputs that declared `3.2` as the default or something. 

If that happens to work, (including with a `range` section like we're doing in python) then I guess I'd suggest doing that, but other wise I'd not bother.

Another option would be to have a pipeline per ruby-version (ruby/require-3.2, ruby/require-3.3).  I suspect that would work, and you could reduce copy/paste code between ruby/require-3.2 and ruby/require-3.3 with nested pipelines.  But if you were doing per-ruby versions i'd want that in wolfi-dev/os rather than melange proper.

So...
that was all long winded way of saying "I think this is fine".